packages/ui-components/src/components/range/range.tsx

```diff
@@ -0,0 +1,92 @@
+import { Component, Host, h, Prop, Element, EventEmitter, Event, State, Watch } from '@stencil/core';
+import { IRange, IRangeEvents } from './range.types';
+
+@Component({
+	tag: 'kv-range',
+	styleUrl: 'range.scss',
+	shadow: true
+})
+export class KvRange implements IRange, IRangeEvents {
+	@Element() el!: HTMLKvRangeElement;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) min: number;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) max: number;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) value?: number = 0;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) step?: number = 1;
+	/** @inheritdoc */
+	@Event() valueChange: EventEmitter<number>;
+
+	@Watch('value')
+	valueChangeHandler(newValue: number) {
+		this._value = newValue;
+	}
+
+	/** Internal value state */
+	@State() _value: number = this.value;
+
+	/** Internal state to set initial button position */
+	@State() initialStateSet: boolean = false;
+
+	componentDidRender() {
+		const slider = this.el.shadowRoot.getElementById('slider') as HTMLElement;
+
+		if (!this.initialStateSet) {
+			this.applyCssStyles(false);
+		}
+
+		slider.oninput = () => {
+			this.applyCssStyles(true);
+		};
+	}
+
+	private applyCssStyles = (launchValueChangeEvent: boolean) => {
+		const rangeInputValue = this.el.shadowRoot.querySelector('input');
+		const inputElement = this.el.shadowRoot.querySelector('input') as HTMLElement;
+		const selector = this.el.shadowRoot.getElementById('selector');
+
+		let inputValue = rangeInputValue.value;
+		const percentage = this.getInputPercentageFromValue(inputValue);
+		const offSet = this.getOffset(percentage);
+
+		selector.style.left = percentage + '%';
+		selector.style.marginLeft = offSet + 'px';
+		inputElement.style.background = `linear-gradient(90deg, #005cc7 ${percentage}%, #121212 ${percentage + 0.1}%)`;
```

Comment:
Hum, probably this could be moved to a `getInputBackground(percentage: number): string` helper

Reply:
and you have another problem you can't have fixed colors that way the component  will never adapt to the theme

Reply:
@ricardogoncalves89 could we consider receiving this two colors as props? Like "backgroundColor: string; sliderColor: string"?

Reply:
Suggestion: Use css props to manipulate that from the implementation side, if needed you can then use JS to manipulate the css props as you see fit.

Something to help: https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty

Let me know if you need any help ;)

Reply:
answered in last review.